### PR TITLE
Improve output of kernel errors in PMacc with PMACC_BLOCKING_KERNEL=ON

### PR DESCRIPTION
**Pull request by sbastrakov**
Now in this mode it catches exceptions that cupla + alpaka can throw, prints kernel information and re-throws.

This change should hopefully help debugging crashes like #3381 by providing proper kernel name and file + line details.

### CODE REVIEW

File: include/pmacc/eventSystem/events/kernelEvents.hpp [hunk 1]
```diff
@@ -185,6 +182,14 @@ namespace exec
                 std::string( " [" ) + m_kernel.m_file + std::string( ":" ) +
                 std::to_string( m_kernel.m_line ) + std::string( " ]" );
 
+            /* alpaka reports errors as exceptions, in the check mode handle it
+             * here while we still have kernel information
+             */
+#if( PMACC_SYNC_KERNEL  == 1 )
+            try
```
Comment (psychocoderHPC) [inline include/pmacc/eventSystem/events/kernelEvents.hpp:189]: discussed offline: `try catch` should be moved into the macro `CUDA_CHECK_KERNEL_MSG`

Comment (sbastrakov) [reply include/pmacc/eventSystem/events/kernelEvents.hpp]: Implemented and rebased.

File: include/pmacc/cuplaHelper/ValidateCall.hpp [hunk 2]
```diff
@@ -24,11 +24,10 @@
 #pragma once
 
 #include <cupla.hpp>
+
 #include <iostream>
 #include <stdexcept>
 
-namespace pmacc
-{
```
Comment (sbastrakov) [inline include/pmacc/cuplaHelper/ValidateCall.hpp:31]: Note: this is unrelated, just was noticed during the development. Since there are only macros in this file, no sense to open the namespace.

**Review by psychocoderHPC — APPROVED** [review]
